src/scanspec/specs.py

```diff
@@ -404,6 +456,19 @@ def axes(self) -> list[Axis]:  # noqa: D102
         assert set(left_axes) == set(right_axes), f"axes {left_axes} != {right_axes}"
         return left_axes
 
+    def duration(self):  # noqa: D102
+        if self.left.duration() is None and self.right.duration is None:
+            return None
+        else:
+            if self.left.duration() == self.right.duration:
+                return self.right.duration()
+            elif self.left.duration() is None:
+                return self.right.duration()
+            elif self.right.duration():
+                return self.left.duration()
+            else:
+                return VARIABLE_DURATION
```

Comment:
Both or neither should define a duration

```suggestion
        left, right = self.left.duration(), self.right.duration()
        if left == right:
            # They are producing the same duration, whatever it is
            return left
        elif left is None or right is None:
            # They aren't both None, but if one is then raise
            raise ValueError("Only one of left and right defines a duration")
        else:
            # They both exist, but are different, so are variable
            return VARIABLE_DURATION
```

Reply:
Done!